### PR TITLE
feat(sdk): adding `datetime` type

### LINKED ISSUE #2102: Missing `datetime` std type
See https://github.com/winglang/wing/issues/1392

Also, make sure that when JSII libraries are imported we map `Date` to this new type.
See https://github.com/winglang/wing/pull/2095/files#r1167668099

### PR DESCRIPTION
**Pull request by tsuf239**
## Description
closes #2102 
(besides adding the type mapping, opened a separate issue: https://github.com/winglang/wing/issues/3569)

a wing test added, checked both preflight and inflight both in tf-aws and sim.

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Review by hasanaburayyan — COMMENTED** [review]
Looking good so far, I added some nits to the comment docstrings I didnt do all of them but if you could take a look.

File: libs/wingsdk/src/std/datetime.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+   * returns a Datetime represents the same date in utc
+   */
+  toUtc(): Datetime {
+    return new Datetime(new Date(this.timestampMs));
+  }
+
+  /**
+   * returns ISO-8601 string
+   */
+  toIso(): string {
```
Comment (Chriscbr) [inline libs/wingsdk/src/std/datetime.ts:191]: Let's aim for consistency in our public APIs and set a model for others to follow. Please use the public keyword and start docstrings with a capital letter for all methods. (When the codebase follows conventions, the contributor experience is easier for newcomers)

Comment (tsuf239) [reply libs/wingsdk/src/std/datetime.ts:203]: No problem, good to know, can I add those two points to the docs somewhere? what will be the best place?

Comment (Chriscbr) [reply libs/wingsdk/src/std/datetime.ts]: Somewhere here would be good: https://www.winglang.io/contributing/start-here/pull_requests

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**